src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -971,9 +971,22 @@ class RsFnInferenceContext(
 
     private fun inferStructTypeArguments(literal: RsStructLiteral, typeParameters: Substitution) {
         literal.structLiteralBody.structLiteralFieldList.mapNotNull { field ->
-            field.expr?.let { expr ->
-                val fieldType = field.type
-                expr.inferTypeCoercableTo(fieldType.substitute(typeParameters))
+            val fieldType = field.type.substitute(typeParameters)
+
+            if (field.expr != null) {
+                field.expr!!.inferTypeCoercableTo(fieldType)
```

Comment:
Please, avoid `!!` in kotlin code. You can extract `field.expr` to a variable and then check `expr != null`

Reply:
Done.